systemd/escape.go

```diff
@@ -0,0 +1,65 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package systemd
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"strings"
+)
+
+const allowed = `:_.abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789`
+
+// EscapePath works like systemd-escape --path
+// FIXME: use github.com/coreos/go-systemd/unit/escape.go
+//        once we can update go-systemd. we can not right now
+//        because versions >= 2 are not compatible with go1.3 from
+//        15.04
+func EscapePath(in string) string {
+	buf := bytes.NewBuffer(nil)
```

Comment:
Thanks! I'm happy to change it if thats better style. The uses in fmt.Fprintf() will also have to be adjusted as they expect a pointer but thats trivial of course.